napari/_qt/layer_controls/_tests/test_qt_labels_layer.py

```diff
@@ -22,6 +22,25 @@ def test_changing_layer_color_mode_updates_combo_box(qtbot):
     assert layer.color_mode == qtctrl.colorModeComboBox.currentText()
 
 
+def test_changing_show_selected_updates_backend(qtbot):
+    """Test that changing show_selected_labels on UI updates the backend
+    and changing on the backend updates the UI"""
+    layer = Labels(_LABELS, color=_COLOR)
+    qtctrl = QtLabelsControls(layer)
+    qtbot.addWidget(qtctrl)
+
+    assert not qtctrl.selectedColorCheckbox.isChecked()
+    assert not layer.show_selected_label
+
+    # ensure setting UI checkbox changes backend
+    qtctrl.selectedColorCheckbox.setChecked(True)
```

Comment:
```suggestion
    with qtbot.wait_signal(qtctrl.selectedColorCheckbox.stateChanged):
        qtctrl.selectedColorCheckbox.setChecked(True)
```
you need to ensure that the change state properly propagates.

Reply:
ah. thank you!